StripeCore/StripeCore/Source/DownloadManager/DownloadManager.swift

```diff
@@ -93,19 +93,34 @@ extension DownloadManager {
         let blockingDownloadSemaphore = DispatchSemaphore(value: 0)
 
         let urlRequest = URLRequest(url: url, cachePolicy: .useProtocolCachePolicy)
-        let task = self.session.downloadTask(with: url) { tempURL, response, _ in
-            guard let tempURL = tempURL,
-                let response = response,
-                let data = self.getDataFromURL(tempURL),
-                let image = self.persistToMemory(data, forImageName: imageName)
-            else {
-                let errorAnalytic = ErrorAnalytic(event: .unexpectedStripeCoreDownloadManagerError,
-                                                  error: Error.downloadSyncFailure,
+        let task = self.session.downloadTask(with: url) { tempURL, response, responseError in
+            if let responseError = responseError as? NSError {
+                blockingDownloadSemaphore.signal()
+                let errorAnalytic = ErrorAnalytic(event: .stripeCoreDownloadManagerError,
+                                                  error: responseError,
+                                                  additionalNonPIIParams: ["url": url.absoluteString])
+                STPAnalyticsClient.sharedClient.log(analytic: errorAnalytic)
+                return
+            }
+
+            guard let tempURL = tempURL, let response = response, let data = self.getDataFromURL(tempURL) else {
```

Comment:
```suggestion
            guard let tempURL, let response, let data = self.getDataFromURL(tempURL) else {
```

Reply:
We should really use the modern `async throw` APIs so that we're guaranteed a tempURL and response if there's no error, among other things.

Reply:
Also why are we using a downloadTask instead of dataTask for this? These images are tiny and we're not downloading them in the background, any reason we shouldn't switch to dataTask and just get the data directly instead of having to read the data from disk? cc @wooj-stripe

Reply:
Refactored this class to use async functions and updated to use a data task.